Putnamos aka "i made LINDA fast"

Putnam3145 opened the pull request:
## About The Pull Request

Made LINDA work by just mixing a tile with its neighbors instantly and fully instead of all this wacky "share" bull. Should make things way, way faster, in many ways.

**TESTMERGE THIS OH MY GOD**

## Why It's Good For The Game

LINDA is super bad. It takes this baffling perfect approach to simulation when good-enough is... good enough, and much faster. This is good-enough.

## Changelog
:cl:
tweak: Putnamos: faster faster faster.
/:cl:

Putnam3145 committed:
putnamos aka "i made linda fast"

Putnam3145 commented on code/__DEFINES/atmospherics.dm:
```diff
@@ -149,9 +149,9 @@
 //OPEN TURF ATMOS
 #define OPENTURF_DEFAULT_ATMOS		"o2=22;n2=82;TEMP=293.15" //the default air mix that open turfs spawn
-#define TCOMMS_ATMOS				"n2=100;TEMP=80" //-193,15�C telecommunications. also used for xenobiology slime killrooms
+#define TCOMMS_ATMOS				"n2=100;TEMP=80" //-193,15�C telecommunications. also used for xenobiology slime killrooms
```
god DAMMIT

Putnam3145 committed:
i need sleep

Ghommie commented on code/__DEFINES/atmospherics.dm:
```diff
@@ -149,9 +149,9 @@
 //OPEN TURF ATMOS
 #define OPENTURF_DEFAULT_ATMOS		"o2=22;n2=82;TEMP=293.15" //the default air mix that open turfs spawn
-#define TCOMMS_ATMOS				"n2=100;TEMP=80" //-193,15�C telecommunications. also used for xenobiology slime killrooms
+#define TCOMMS_ATMOS				"n2=100;TEMP=80" //-193,15�C telecommunications. also used for xenobiology slime killrooms
```
```suggestion
#define TCOMMS_ATMOS				"n2=100;TEMP=80" //-193,15°C telecommunications. also used for xenobiology slime killrooms
```
Deja vu.

Putnam3145 replied:
can we just replace it with the word "degrees"

Ghommie commented on code/__DEFINES/atmospherics.dm:
```diff
@@ -149,9 +149,9 @@
+#define TCOMMS_ATMOS				"n2=100;TEMP=80" //-193,15�C telecommunications. also used for xenobiology slime killrooms
 #define AIRLESS_ATMOS				"TEMP=2.7" //space
-#define FROZEN_ATMOS				"o2=22;n2=82;TEMP=180" //-93.15�C snow and ice turfs
+#define FROZEN_ATMOS				"o2=22;n2=82;TEMP=180" //-93.15�C snow and ice turfs
```
```suggestion
#define FROZEN_ATMOS				"o2=22;n2=82;TEMP=180" //-93.15°C snow and ice turfs
```

Ghommie commented on code/modules/atmospherics/environmental/LINDA_turf_tile.dm:
```diff
@@ -199,30 +209,43 @@
+			var/datum/gas_mixture/G = new
+			G.copy_from_turf(src)
+			ARCHIVE_TEMPERATURE(G)
+			if(!(our_air.compare(G)))
```
Are the parethesis after the ! operator really needed?

Poojawa replied:
they don't hurt for sure.

Ghommie commented on code/modules/atmospherics/environmental/LINDA_turf_tile.dm:
```diff
@@ -154,28 +157,35 @@
+		var/datum/gas_mixture/G = new
+		G.copy_from_turf(src)
+		ARCHIVE_TEMPERATURE(G)
+		if(!(our_air.compare(G)))
```
Same.

Ghommie reviewed (changes requested):
Minor things, not a ss13 atmos code expert.

Putnam3145 committed:
yeah here's some updates

Putnam3145 committed:
Properly named some coefficients, made them runtime adjustable

Poojawa commented on code/controllers/subsystem/air.dm:
```diff
@@ -44,6 +45,12 @@ SUBSYSTEM_DEF(air)
 	var/list/queued_for_activation
 
+	// VORELINDA config options (am I really calling it that?)
+	// The below two refer to how many SSair_turf ticks will go by before a turf will run again.
```
I'm getting prob(80) vibes

Ghommie replied:
Thing is `prob(80)` has no fixed, consistent (per tick) frequency, unlike here.

deathride58 replied:
This is there primarily to artificially slow down the simulation. Without it comes quite a few gameplay issues that come from TG's map design... *really* not being meant for ZAS propagation speeds

Putnam3145 committed:
yikes

Ghommie commented on code/controllers/subsystem/air.dm:
```diff
@@ -44,6 +45,12 @@ SUBSYSTEM_DEF(air)
 	var/map_loading = TRUE
 	var/list/queued_for_activation
 
+	// VORELINDA config options (am I really calling it that?)
```
> VORELINDA

Any other name that doesn't explicitly reference that weird fetish, like LINDA+, V-LINDA, PUT-LINDA, P-LINDA, CIT-LINDA, would be appreciated.

lolman360 replied:
LINDA-next

Putnam3145 committed:
locales continue to have been bad

Co-Authored-By: Ghom <42542238+Ghommie@users.noreply.github.com>

Putnam3145 committed:
like seriously

Co-Authored-By: Ghom <42542238+Ghommie@users.noreply.github.com>

Putnam3145 committed:
deg.

Putnam3145 committed:
Merge branch 'fastmos' of https://github.com/Putnam3145/Citadel-Station-13 into fastmos

Putnam3145 committed:
renamed it

Putnam3145 committed:
made spacewind, space more dangerous

Putnam3145 committed:
Removed blocksTemperature.

it's... the same thing as BlocksSuperconductivity lol.

Putnam3145 committed:
tweaks.

Putnam3145 committed:
Merge branch 'master' of https://github.com/Citadel-Station-13/Citadel-Station-13 into fastmos

Putnam3145 committed:
GASGASGAS

Putnam3145 committed:
put on the GASGASGAS

Putnam3145 committed:
Groups are breaking down too easy

Putnam3145 committed:
Merge branch 'master' of https://github.com/Citadel-Station-13/Citadel-Station-13 into fastmos

Putnam3145 committed:
Updated passive vent/relief valve to work more... fastmos-like

Putnam3145 committed:
Merge branch 'master' of https://github.com/Citadel-Station-13/Citadel-Station-13 into fastmos

Putnam3145 committed:
wow thanks atmos very cool

Putnam3145 committed:
Merge branch 'master' of https://github.com/Citadel-Station-13/Citadel-Station-13 into fastmos

Putnam3145 committed:
blocksTemperature should never have been a thing

Putnam3145 committed:
removed planetary atmos (for now)

Putnam3145 committed:
what!!!

Putnam3145 committed:
don't panic

Putnam3145 committed:
parentheses for clarity

Putnam3145 committed:
lowering the cost of GASGASGAS

Putnam3145 committed:
Made garbage collection WAY more zealous.

Putnam3145 committed:
Merge branch 'master' of https://github.com/Citadel-Station-13/Citadel-Station-13 into fastmos

Putnam3145 committed:
Makes space air not immediately fully coldify all surroundings

Putnam3145 committed:
squawk

Putnam3145 committed:
Made coolant gases stronger for supermatter.

This is to make up for unforeseen consequences of having the gases update more often: rather than gases moving from one side of the supermatter to the other, they spread out instantly, which makes the heat spread much faster. To counteract this, I'm making coolant gases significantly stronger. This may need to be dialed back, but for now, it's a fix.

Putnam3145 committed:
Brought suspension ratio back to normal
